StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/ElementOrderCodeFixProvider.cs

```diff
@@ -0,0 +1,180 @@
+            }
+
+            if (parentDeclaration is NamespaceDeclarationSyntax)
+            {
+                return HandleNamespaceDeclaration(memberToMove, (NamespaceDeclarationSyntax)parentDeclaration, checks, syntaxRoot);
+            }
+
+            if (parentDeclaration is CompilationUnitSyntax)
+            {
+                return HandleCompilationUnitDeclaration(memberToMove, (CompilationUnitSyntax)parentDeclaration, checks, syntaxRoot);
+            }
+
+            return syntaxRoot;
+        }
+
+        private static SyntaxNode HandleTypeDeclaration(MemberOrderHelper memberOrder, TypeDeclarationSyntax typeDeclarationNode, ElementOrderingChecks checks, SyntaxNode syntaxRoot)
+        {
+            return OrderMember(memberOrder, typeDeclarationNode.Members, checks, syntaxRoot);
+        }
+
+        private static SyntaxNode HandleCompilationUnitDeclaration(MemberOrderHelper memberOrder, CompilationUnitSyntax compilationUnitDeclaration, ElementOrderingChecks checks, SyntaxNode syntaxRoot)
+        {
+            return OrderMember(memberOrder, compilationUnitDeclaration.Members, checks, syntaxRoot);
+        }
+
+        private static SyntaxNode HandleNamespaceDeclaration(MemberOrderHelper memberOrder, NamespaceDeclarationSyntax namespaceDeclaration, ElementOrderingChecks checks, SyntaxNode syntaxRoot)
+        {
+            return OrderMember(memberOrder, namespaceDeclaration.Members, checks, syntaxRoot);
+        }
+
+        private static SyntaxNode OrderMember(MemberOrderHelper memberOrder, SyntaxList<MemberDeclarationSyntax> members, ElementOrderingChecks checks, SyntaxNode syntaxRoot)
+        {
+            foreach (var member in members)
+            {
+                var orderHelper = new MemberOrderHelper(member, checks);
+
+                if (orderHelper.Priority < memberOrder.Priority)
+                {
+                    syntaxRoot = MoveMember(syntaxRoot, memberOrder.Member, member);
+                    break;
+                }
+            }
+
+            return syntaxRoot;
+        }
+
+        private static SyntaxNode MoveMember(SyntaxNode root, MemberDeclarationSyntax member, MemberDeclarationSyntax targetMember)
+        {
+            var trackedRoot = root.TrackNodes(member, targetMember);
+            var memberToMove = trackedRoot.GetCurrentNode(member);
+            var targetMemberTracked = trackedRoot.GetCurrentNode(targetMember);
+            if (!memberToMove.HasLeadingTrivia)
+            {
+                memberToMove = memberToMove.WithLeadingTrivia(targetMemberTracked.GetLeadingTrivia().Where(x => x.IsKind(SyntaxKind.WhitespaceTrivia)).LastOrDefault());
+            }
+
+            root = trackedRoot.InsertNodesBefore(targetMemberTracked, new[] { memberToMove });
+            var fieldToMoveTracked = root.GetCurrentNodes(member).Last();
+            return root.RemoveNode(fieldToMoveTracked, SyntaxRemoveOptions.KeepNoTrivia);
+        }
```

Comment:
As noted for several test cases above, this code should preserve empty lines between members, if the members are not defined on a single line. Otherwise this code fix will trigger SA1516 diagnostics after fixing, which should not happen imho.